readthedocs/projects/tasks.py

```diff
@@ -625,10 +625,15 @@ def get_env_vars(self):
                     self.project.doc_path,
                     'envs',
                     self.version.slug,
-                    'bin'
+                    'bin',
                 ),
             })
 
+        # Update environment from Project's specific environment variables
+        env.update(
+            api_v2.project(self.project.pk).environment_variables().get(),
+        )
```

Comment:
Heroku limits variables from starting with "HEROKU_", should we do similar? I guess worst case the user is just breaking their own builds, there isn't anything internal we use the env vars for -- I think at least.

Reply:
We could.

They have more validations: https://devcenter.heroku.com/articles/config-vars#config-var-policies

The one that limits the 32kb is a good one, since if you have more data there it's very likely that you are trying to do some "weird thing".

Anyway, I think this validation should go in the Form when we allow the user to create vars from the admin tab.

Reply:
I'm solidly on the fence here, so feel free to take this any direction you want. We probably don't need to be too strict here.

There might be some wisdom to something like the 32kb limit (like, does docker choke on that many environment variables? I wouldn't put too much work in here, so I'm :+1: on adding anything that is easy to implement and stops users from breaking things.

Reply:
Let do this validation on the form then, which is the easier place to implement and it will be per- env variable instead of all together.

Reply:
I've already started writing this form in another PR: https://github.com/rtfd/readthedocs.org/pull/4899/files#diff-49158d3dbeadb679e46430b842c441a6R786

Reply:
Cool, can push discussion there!